### ISSUE TITLE
Add argument to checkpoint every *n* epochs in references/classification/train.py

### ISSUE THREAD

**Issue by JonnaMat** [pytorch/vision#7481]
### 🚀 The feature

In the training script mentioned in the title, the model + training is checkpointed after each epoch if `--output-dir` is provided. 
I would like to add an argument which gives the option of checkpointing every *n* epochs.,

### Motivation, pitch

As a DL researcher, I am using this script for training several models on ImageNet. As I don't want to checkpoint after each epoch I would like more flexibility.

**Comment by NicolasHug** [comment]
Hi @JonnaMat ,

That's a reasonable feature request. Out of curiosity, how are you using those references scripts? Do you just checkout the torchvision repo and use them as-is, or do you make a few changes before that?

The reason I'm asking is because we intend those training references to be copy/paste-able and adaptable to a wide range of use-cases, but the corollary is that they might not always be "plug and play" straight out of the box - and that's OK.

We could add a new parameter to support your use-case, but because we don't offer any backward compatibility guarantee on that part of the code base, it may not be around forever.

Basically: if you submit a PR to add it we'll review it, but we also want to empower you to do those changes locally and adapt the references to suit your needs.

**Comment by JonnaMat** [comment]
Hi @NicolasHug ,

Thanks for your reply and thanks for clarifying that the mentioned script might not always be "plug and play"! For my current use case, I found the script to cover mostly everything - thus, I only intend to locally add minor changes to the script. 

Since the feature of not checkpointing after each epoch might be relevant to others as well, I will go ahead a submit a PR. :)
